### PR TITLE
Move more XContent.createParser calls to non-deprecated version

### PR DESCRIPTION
**Pull request by dakrone**
This moves more of the callers to pass in the DeprecationHandler.

Relates to #28504

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTermsAggregationBuilder.java [hunk 1]
```diff
@@ -96,7 +97,7 @@
             aggregationParser.declareObject(SignificantTermsAggregationBuilder::significanceHeuristic,
                     (p, context) -> {
                         SignificanceHeuristicParser significanceHeuristicParser = significanceHeuristicParserRegistry
-                                .lookupReturningNullIfNotFound(name);
+                                .lookupReturningNullIfNotFound(name, LoggingDeprecationHandler.INSTANCE);
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTermsAggregationBuilder.java]: I think this should be `p.getDeprecationHandler()`.

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTextAggregationBuilder.java [hunk 2]
```diff
@@ -111,7 +112,7 @@
             PARSER.declareObject(SignificantTextAggregationBuilder::significanceHeuristic,
                     (p, context) -> {
                         SignificanceHeuristicParser significanceHeuristicParser = significanceHeuristicParserRegistry
-                                .lookupReturningNullIfNotFound(name);
+                                .lookupReturningNullIfNotFound(name, LoggingDeprecationHandler.INSTANCE);
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTextAggregationBuilder.java]: Same.

File: server/src/main/java/org/elasticsearch/search/aggregations/pipeline/movavg/MovAvgPipelineAggregationBuilder.java [hunk 3]
```diff
@@ -402,7 +403,8 @@ public static MovAvgPipelineAggregationBuilder parse(
             factory.predict(predict);
         }
         if (model != null) {
-            MovAvgModel.AbstractModelParser modelParser = movingAverageMdelParserRegistry.lookup(model, parser.getTokenLocation());
+            MovAvgModel.AbstractModelParser modelParser = movingAverageMdelParserRegistry.lookup(model,
+                parser.getTokenLocation(), LoggingDeprecationHandler.INSTANCE);
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/search/aggregations/pipeline/movavg/MovAvgPipelineAggregationBuilder.java]: `parser.getDeprecationHandler()`?

**Review by nik9000 — CHANGES_REQUESTED** [review]
I think I found a few things.

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTermsAggregationBuilder.java (hunk 1 above)
Comment (dakrone) [reply server/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTermsAggregationBuilder.java]: Whoops yeah, I missed that `p` was a parser

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTextAggregationBuilder.java (hunk 2 above)
Comment (dakrone) [reply server/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTextAggregationBuilder.java]: Should be fixed now

File: server/src/main/java/org/elasticsearch/search/aggregations/pipeline/movavg/MovAvgPipelineAggregationBuilder.java (hunk 3 above)
Comment (dakrone) [reply server/src/main/java/org/elasticsearch/search/aggregations/pipeline/movavg/MovAvgPipelineAggregationBuilder.java]: Yep, good catch

**Comment by dakrone** [comment]
Thanks for looking @nik9000, I pushed a commit

**Review by nik9000 — APPROVED** [review]

**Pull request merged**